### PR TITLE
[WIP] Adds Overmap Special: Nursing Home

### PR DESCRIPTION
**Pull request by pshepjr**
#### Summary
Content "Adds Overmap Special, Nursing Home"

#### Purpose of change

This PR adds a new overmap special, the Nursing Home.

#### Describe the solution

This location is intended to be a mid-to-end game location. At first, it doesn't seem to have anything terribly valuable - perhaps some medicine, food, and basic items. However, a secret lab is hidden underneath. If they defeat the boss encounter, the survivor can find a trove of mutagens and CBMs.

For those who are interested, there are some story elements such as consoles with emails/documents that can be read.

It is a Unique overmap special (50% occurrence) that typically spawns close to, or even next to, a city.

I've expanded more in detail below.

#### Describe alternatives you've considered

None

#### Testing

Many, many game restarts checking to see if it spawned, and force spawning in the console.

#### Additional context

**MONSTERS ADDED**

Zeriatric:
> "A hunched, decrepit, geriatric zombie. Its dentures seem to be falling out yet you swear you can still hear them mumbling about the good old days."
> 
> Zeriatrics are weak, frail, slow, and deaf. They also can't bite you. Their vision range is extremely poor, even in the daylight. All in all, they aren't much of a threat. That being said - when excited, they tend to spit their dentures out. And those suckers can hurt.
> 
> Loot consists of basic clothing, with a chance of glasses and/or a cane.

Feral Granny:

> "A sweet old lady in a wheelchair. As you move closer, she offers you some cookies. Do you want a cookie?"
> 
> Feral Grannies are exactly as the name would imply. Weaker and frailer than their feral human counterparts, they throw stale cookies at the survivor while chasing them down in a wheelchair (the damage of the cookies are really low, likely bouncing off any decent armor - its mostly for flavor). Because of their higher speed, they can be difficult to run away from. They tend to grab on to the survivor for dear life while wailing and spouting nonsense, attracting other Zeds. Their unpulped corpse will become a crawling zombie.
> 
> Loot consists of basic clothing, their wheelchair, and a chance of glasses and/or a cane. Additionally, they have a low chance of dropping narcotics (story reason).

Big Pappy:

> "A gigantic, monstrous old man with a crazed, anguished face. Created by a twisted science experiment. Carries a large cane, and seems intent to discipline you with it."
> 
> A mutant/cyborg resulting from unethical experimentation. When the survivor finds the secret lab and reaches the end of it, Big Pappy breaks out from his cell and attacks the survivor unrelentingly, flanked by two Feral Grannies. Stat wise, Big Pappy is the feral version of a zombie wrestler, but with some additional adjustments. To quote him, "I'm 'bout to walk 15 miles both ways across your ass." Big Pappy is fully intended to be more of a boss monster and not something to be easily brushed off.
> 
> Loot consists of a bathrobe and footrags. Big Pappy will drop his cane - a unique weapon that is essentially a durable ironshod quarterstaff with higher damage. If dissected, he shares a CBM loot table with the prototype cyborg.
> 

Other enemies found here: Regular zombies, fat zombies, crawlers, Boomers (because obviously..), medical zombies, scientist zombies [in the lab], technician zombies [in the lab].

**ITEMS/VEHICLES ADDED**

Small Hospital Bed: It's the hospital bed, but only one tile.

Folded Folding Wheelchair: Lootable item that deploys the existing Folding Wheelchair vehicle. This is part of the Feral Granny loot table.

Big Pappy's Cane: Quarterstaff weapon dropped by Big Pappy.

**STORY**

Shady Hills nursing home - at first glance to any survivor, it's an average nursing home location with mediocre loot. It can be somewhat dangerous to those unprepared, as all of the entrances save the front door are locked metal doors, and the windows are all reinforced. While most of the monsters aren't terribly strong, these features can essentially can turn the place into a death trap if one isn't careful.

Random pieces of trash can be found on the floor. There is a suspiciously high number of frozen junk food found in the kitchen. Were these poor elderly folks taken care of well at all?

Various computers can be found in the building. The survivor can read various emails between staff and documents stored on the drive. There are heavy implications that many employees weren't happy working here, there are a high number of lawsuits filed against the home, and certain employees are certainly up to something shady. Apparently, one of the residents even went missing. How does a resident go missing!?

A perceptive enough survivor [in fairness it isn't too hard] will find a hidden door to an underground lab. Then it becomes clear where that missing resident ended up...

### CODE REVIEW

**Comment by Zireael07** [comment]
Hmm, personally I would just stick with Zeriatric (nice name btw), the other two strike me as more fitting for Crazy Cata mod.

**Review by Venera3 — COMMENTED** [review]
A few points:
- One-stop shop labs (labs with every type of labby loot) are one the way out, and 50% spawn chance per overmap is too high for one in any case. 
- The tone of the monsters is very jarring - spitting dentures and throwing cookies is definitely Crazy Cataclysm level, and zombies don't mumble for one thing
- Pappy is a new cyborg, and terrestrial bionics are on the way out. The magic cane is also a bit on the gamey side.

Changes I'd recommend:
- Lower the spawn chance to be more in line with the other rare lab variants (prison labs, for example) and add a version without the lab - a mundane nursing home is definitely the kind of location we could use more of
- Instead of hiding a lab (accessible through a tiny stair and not connected to the subways) you could instead have a hidden XEDRA "voluteer recruitment" area and maybe a transport truck with handcuffs and a TCL trucker badge out back. That also works better with the "disappearing residents" story - a single nursing home won't give you enough subjects to justify a lab without somebody noticing, but a few yoinked grandmas could work.

**Comment by bombasticSlacks** [comment]
So the devs talked about it a bit last night and we think the idea is fun and also a lot of work has gone into it. 

What we think needs to happen for this to be good to go is:

- There should be a non lab variant of this special. Can be basically the same layout, just without the lore and with the ferals replaced with normal zombies (since no one is working on them). Zeriatrics could reasonably still be there.
- That should spawn most of the time (I would think >95%) with this version being <5%
- Then nursing homes can be more ubiquitous (as they should be) 
- Then it's a surprise and actually interesting when you find the weird one


As far as loot goes people have already mentioned but the cybernetics and mutagen aren't really a great fit:

- The cybernetics are getting phased out so that's a hard no
- The mutatagen is so ubiquitous for end game locales it IMO doesn't really encourage exploring for it and it makes it samey with all the other locations
- They aren't really mutants
- This is a personal point but, you really only need to loot a certain amount of mutagen per character. Once you've got your mutant it isn't exciting to find any more mutagen because you aren't going to use it.

There is a current push for new #53141 super soldier serums which this would be fitting for. Without those existing though I think having a big supply of https://nornagon.github.io/cdda-guide/#/item/ampoule which goes in the RX12 jet injector would be a lot of fun. This means it also makes this locations loot a lot more unique than just being one more spot to find mutagen. It also from what I saw from the snippets matches up with the theming (living forever, being super aggro even if old etc.)


For the super cane. We try to avoid things that are in-explainable oddities. So the cane does a lot of damage but why? Is it just a big stick, if so why couldn't the player just make a similar big stick? So for this I would say either pair it back to be just a simple cane or make it really weird so it falls into the realm of fiction. 
*my personal recommendation* (I haven't ran this by the rest of the team yet) is it could actually be a part of big pappi himself (like a distended hardened bone he's ripped out of himself). It could have similar rules to the migo armor plates, so as you use it as a weapon it can take damage and over time grows back. It could start doing mostly bash damage (being a big club) and as it wears down and fragments become lighter and do more pierce. Something like that. This is just spit balling but if you wanted me to write the code for such a weapon I would be happy to contribute it.

**Comment by pshepjr** [comment]
Hi there, thanks for the feedback! Here is what I am thinking below, let me know your thoughts when you can!

- A non lab variant is a good thought, I will do that. This actually started as just that and then my creativity carried me away and that's when I thought, "oh, having this spawn multiple times in an overmap wouldn't make sense."
- I didn't realize cybernetics were being phased out, clearly I am behind the times! And good points about the mutagen.
- I can remove the super cane and replace his loot table with a regular cane.  

For the ranged attacks possessed by the Zeriatrics and Feral Grannies, I would request the dev team's final decision if they feel this is not appropriate. I will certainly remove those attacks if it is felt they do not fit. However, here are my respectful counter-arguments to that point:

- For grannies:  My understanding of the ferals is that they are basically insane. Feral Human enemies throw rocks (arguably, random rocks that don't always make sense for them to have, depending on location), so my logic is that a feral granny would likely have a very similar idea - "throw something at this thing to kill it". In the granny's case, they don't have a rock. So they throw their cookies. While it is certainly comical, I don't see it as necessarily not making sense. A common 'grandmotherly stereotype' is baking cookies, after all.

- For zeriatrics: Zombies clearly can have the capacity to spit, as seen by other variants (boomers, etc), and if the zombie had something else in its mouth, I don't see it being a stretch that it may try to spit it at something it is trying to kill. Also - just to be clear, they can only do it once (its coded to have an ammo supply of 1), and it's also why I removed their ability to bite. It's also a relatively short range (4). 

Again, I mean these arguments respectfully, so if the dev team feels they don't fit I will certainly remove them. 

I may rework this for now to cut out the lab entirely and leave just the top floor, including the zeriatrics and grannies (cutting Big Pappy entirely). Let me know if this sounds like a reasonable step. I like the concept of the super soldier serums - perhaps I can revisit the lab part when those are released. Nothing wrong with the nursing home just being badly managed in "the good old days" :)

Thank you again!

**Comment by bombasticSlacks** [comment]
>I may rework this for now to cut out the lab entirely and leave just the top floor, including the zeriatrics and grannies (cutting Big Pappy entirely). Let me know if this sounds like a reasonable step. I like the concept of the super soldier serums - perhaps I can revisit the lab part when those are released. Nothing wrong with the nursing home just being badly managed in "the good old days" :)

My only thought would be the work is already done, so having it as a fun weird location to stumble upon might be nice. A lot of DDA players favorite moments are a time they found something they weren't expecting. Even having it without a clear reward (other than assorted meds) is perfectly fine and again I think jet injector would be a good holdover, but in this case do what you think is right.

**Comment by pshepjr** [comment]
> > I may rework this for now to cut out the lab entirely and leave just the top floor, including the zeriatrics and grannies (cutting Big Pappy entirely). Let me know if this sounds like a reasonable step. I like the concept of the super soldier serums - perhaps I can revisit the lab part when those are released. Nothing wrong with the nursing home just being badly managed in "the good old days" :)
> 
> My only thought would be the work is already done, so having it as a fun weird location to stumble upon might be nice. A lot of DDA players favorite moments are a time they found something they weren't expecting. Even having it without a clear reward (other than assorted meds) is perfectly fine and again I think jet injector would be a good holdover, but in this case do what you think is right.

Thank you for your thoughts! 

How about this - as suggested earlier, we have a common variant without the lab (keeping zeriatrics and grannies). Then a rare special with the lab, using the meds as you suggested. Later on, possibly adding some of the new supersoldier serums.

One additional question - should autodocs and autoclaves also be removed? 

Thanks!

**Comment by bombasticSlacks** [comment]
>How about this - as suggested earlier, we have a common variant without the lab (keeping zeriatrics and grannies). Then a rare special with the lab, have the meds as loot, per your suggestion. Later on, possibly adding some of the new supersoldier serums.

I think this would be good

> One additional question - should autodocs and autoclaves also be removed?

yes

File: data/json/monsters/feral_humans.json [hunk 1]
```diff
@@ -1019,5 +1019,140 @@
       "PATH_AVOID_DANGER_1",
       "DROPS_AMMO"
     ]
+  },
+  {
+    "id": "mon_feral_big_pappy",
+    "type": "MONSTER",
+    "name": { "str": "Big Pappy" },
+    "//": "Boss enemy for Nursing Home special",
+    "description": "A gigantic, monstrous old man with a crazed, anguished face. Created by a twisted science experiment. Carries a large cane, and seems intent to discipline you with it.",
```
Comment (TheShadowFerret) [inline data/json/monsters/feral_humans.json:1028]: Two sentence fragments bad
```suggestion
    "description": "A gigantic, monstrous old man with a crazed, anguished face created by a twisted science experiment. He carries a large cane, and seems intent to discipline you with it.",
```

**Pull request closed without merging**